From 15e5d86cc55ad94ac946e04d5e25b84be5810970 Mon Sep 17 00:00:00 2001
From: Brett T. Warden <brett.t.warden@intel.com>
Date: Fri, 30 Dec 2022 08:43:25 +0100
Subject: [PATCH] Remove dependency on BitArray.h from zxing-1.2.0

In zxing-1.4.0, numerous headers are no longer public. Rework the
ConvertToSVGFormat method so it uses bitmatrix.get instead of
bitmatrix.getRow, similar to the ToSVG method in zxing itself.

See https://github.com/zxing-cpp/zxing-cpp/issues/361

Change-Id: Ie25eb8f782e8799fbd57c24ef79bba92acf0f9ff
Reviewed-on: https://gerrit.libreoffice.org/c/core/+/144874
Tested-by: René Engelhard <rene@debian.org>
Reviewed-by: René Engelhard <rene@debian.org>
Tested-by: Jenkins
Reviewed-by: Caolán McNamara <caolanm@redhat.com>
---

diff --git a/cui/source/dialogs/QrCodeGenDialog.cxx b/cui/source/dialogs/QrCodeGenDialog.cxx
index f8cbac1..817be7f 100644
--- a/cui/source/dialogs/QrCodeGenDialog.cxx
+++ b/cui/source/dialogs/QrCodeGenDialog.cxx
@@ -27,7 +27,6 @@
 #endif
 
 #include <BarcodeFormat.h>
-#include <BitArray.h>
 #include <BitMatrix.h>
 #include <MultiFormatWriter.h>
 #include <TextUtfEncoding.h>
@@ -79,7 +78,6 @@
     OStringBuffer sb;
     const int width = bitmatrix.width();
     const int height = bitmatrix.height();
-    ZXing::BitArray row(width);
     sb.append("<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
               "<svg xmlns=\"http://www.w3.org/2000/svg\" version=\"1.1\" viewBox=\"0 0 "
               + OString::number(width) + " " + OString::number(height)
@@ -87,10 +85,9 @@
                 "<path d=\"");
     for (int i = 0; i < height; ++i)
     {
-        bitmatrix.getRow(i, row);
         for (int j = 0; j < width; ++j)
         {
-            if (row.get(j))
+            if (bitmatrix.get(j, i))
             {
                 sb.append("M" + OString::number(j) + "," + OString::number(i) + "h1v1h-1z");
             }
